core/DependencyInjection/ContainerServiceProvider.php

```diff
@@ -45,18 +46,26 @@ public function __invoke(ContainerConfigurator $configurator): void
     {
         $services = $configurator->services();
 
-        $services->set(ServerRequestInterface::class, ServerRequestInterface::class)
+        $services
+            ->set(ServerRequestInterface::class, ServerRequestInterface::class)
             ->public()
             ->factory(ServerRequest::class . '::fromGlobals');
 
-        $services->set(ResponseInterface::class, Response::class)
+        $services
+            ->set(ServiceManager::class, ServiceManager::class)
+            ->public()
+            ->factory(ServiceManager::class . '::getServiceManager');
+
+        $services
+            ->set(ResponseInterface::class, Response::class)
             ->public();
 
         $services->set(RelayBuilder::class, RelayBuilder::class);
 
         $services->set(LegacyServiceGateway::class, LegacyServiceGateway::class);
 
-        $services->set(MiddlewareRequestHandler::class, MiddlewareRequestHandler::class)
+        $services
+            ->set(MiddlewareRequestHandler::class, MiddlewareRequestHandler::class)
```

Comment:
@shpran Do not need to revert if you do not want too, but I personally prefer the other way (one instruction per line) instead of a line wit a floating variable.

I would keep in this PR only the `ServiceManager::class` addition.

Reply:
Done